packages/context-menu/src/vaadin-context-menu.js

```diff
@@ -552,96 +552,50 @@ class ContextMenu extends ElementMixin(ThemePropertyMixin(ItemsMixin(GestureEven
   /** @private */
   __alignOverlayPosition() {
     const overlay = this.$.overlay;
+
+    if (overlay.positionTarget) {
+      // overlay-position-mixin is used
+      return;
+    }
+
     const style = overlay.style;
 
     // Reset all properties before measuring
     ['top', 'right', 'bottom', 'left'].forEach((prop) => style.removeProperty(prop));
     ['right-aligned', 'end-aligned', 'bottom-aligned'].forEach((attr) => overlay.removeAttribute(attr));
 
     // Maximum x and y values are imposed by content size and overlay limits.
-    const { xMax, xMin, yMax, left, right, top, width } = overlay.getBoundaries();
+    const { xMax, xMin, yMax } = overlay.getBoundaries();
     // Reuse saved x and y event values, in order to this method be used async
     // in the `vaadin-overlay-change` which guarantees that overlay is ready
-    let x = this.__x || (!this.__isRTL ? left : right);
-    const y = this.__y || top;
+    let x = this.__x;
+    const y = this.__y;
 
     // Select one overlay corner and move to the event x/y position.
     // Then set styling attrs for flex-aligning the content appropriately.
     const wdthVport = document.documentElement.clientWidth;
     const hghtVport = document.documentElement.clientHeight;
 
-    // Align to the parent menu overlay, if any.
-    const parent = overlay.parentOverlay;
-    let alignedToParent = false;
-    let parentContentRect;
-    if (parent) {
-      parentContentRect = parent.$.overlay.getBoundingClientRect();
-      if (parent.hasAttribute('right-aligned') || parent.hasAttribute('end-aligned')) {
-        const parentStyle = getComputedStyle(parent);
-        const getPadding = (el, direction) => {
-          return parseFloat(getComputedStyle(el.$.content)['padding' + direction]);
-        };
-        const dimensionToSet = parseFloat(parentStyle[this.__isRTL ? 'left' : 'right']) + parentContentRect.width;
-        const padding = getPadding(parent, 'Left') + getPadding(overlay, 'Right');
-
-        // Preserve end-aligned, if possible.
-        if (wdthVport - (dimensionToSet - padding) > width) {
-          this._setEndAligned(overlay);
-          style[this.__isRTL ? 'left' : 'right'] = dimensionToSet + 'px';
-          alignedToParent = true;
-        }
-      } else if (x < parentContentRect.x) {
-        // Check if sub menu opens on the left side and the parent menu is not right aligned.
-        // If so, use actual width of the submenu content instead of the parent menu content.
-        x = x - (width - parentContentRect.width);
-      }
-    }
-
-    if (!alignedToParent) {
-      if (!this.__isRTL) {
+    if (!this.__isRTL) {
```

Comment:
> Is it because the sub-menu positioning is now covered by the mixin, and this now only needs to cover the cases for the root menu?

Yes, this file had a good amount of logic related to sub-menu positioning. All that can be removed.

> Also do you see any opportunities to reuse something from the mixin here (e.g. extract a function from the mixin that can be called here)? The logic should be similar at least.

The logic here is conceptually different from that of the position mixin in that there's no relative root element here, it's just about the placement of the `<vaadin-context-menu>`'s root menu. Maybe if we added a new feature to the overlay-position-mixin to have it positioned by given coordinated, then we could eliminate the rest of the positioning logic here. But for now, that feature would only be used by `<vaadin-context-menu>` so it should be fine to have this code here without references to the mixin.

Reply:
OK, thanks for clarifying. We don't have to reuse every piece of code, just double-checking if there were some simple opportunities.